haskell.nix

```diff
@@ -0,0 +1,108 @@
+# A flake-parts module for Haskell cabal projects.
+{ config, lib, flake-parts-lib, ... }:
+
+let
+  inherit (flake-parts-lib)
+    mkSubmoduleOptions
+    mkPerSystemOption;
+  inherit (lib)
+    mkOption
+    mkDefault
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.haskellProject = {
+          haskellPackages = mkOption {
+            type = types.anything;
```

Comment:
`types.anything` is being used 4 times here. That feels wrong to me. Are there correct ways to define these types?

Reply:
`anything` allows rather a lot and merges recursively, which gets weird with packages.

`attrsOf raw` is a safer choice here.